src/dbus_api/macros.rs

```diff
@@ -6,16 +6,16 @@
 /// Macro for early return with Ok dbus message on failure to get pool.
 macro_rules! get_pool {
     ( $engine:ident; $uuid:ident; $default:expr; $message:expr ) => {
-        match $engine.get_pool(&$uuid) {
-            Some(pool) => pool,
-            None => {
-                let (rc, rs) = code_to_message_items(DbusErrorEnum::POOL_NOTFOUND,
-                                                     format!("no pool for uuid {}", $uuid));
-                return Ok(vec![$message.append3($default, rc, rs)]);
-            }
+        if let Some(pool) = $engine.get_pool(&$uuid) {
+            pool
+        } else {
+            let (rc, rs) = code_to_message_items(DbusErrorEnum::POOL_NOTFOUND,
+                                                 format!("no pool for uuid {}", $uuid));
+            return Ok(vec![$message.append3($default, rc, rs)]);
```

Comment:
I'ld like to hold onto it. It saves a level of indentation, which is nice. I think ```unwrap_or_else``` would be kind of awkward.